OpenSim/Actuators/VandenBogert2011Muscle.h

```diff
@@ -0,0 +1,568 @@
+        ImplicitResidual calcJacobianByFiniteDiff(SimTK::Vec2 y,
+                                                  SimTK::Vec2 ydot,
+                                                  double muscleLength,
+                                                  double u, double h ) const;
+        /** Calculate Jacobian by finite difference
+              @param muscleLength Muscle Length
+              @param projFibLenNorm The projected fiber length (inline with
+              tendon) and normalized by muscle optimal length.
+              @param activ The muscle activation
+              @param projFibVelNorm The projected fiber velocity (inline with
+              tendon) and normalized by muscle optimal length.
+              @param activdot The time derivative of the muscle activation
+              @param u The muscle control (excitation)
+              @param h The peturbation size of the finite step
+              @returns TBD for now see ImplicitResidual structure*/
+        ImplicitResidual calcJacobianByFiniteDiff(double muscleLength,
+                                                  double projFibLenNorm,
+                                                  double activ,
+                                                  double projFibVelNorm,
+                                                  double activdot, double u,
+                                                  double h) const;
+
+
+
+
+        /** Calculate the muscle static equilibrium.  The muscle length is
+            static, the fiber velocity is zero, and the activation derivative is
+            zero.
+                @param muscleLength The muscle length
+                @param activ The muscle activation
+                @returns Vector:
+                        \li 0: projected fiber length normalized
+                        \li 1: muscle force */
+        SimTK::Vec2 calcFiberStaticEquilbirum(double muscleLength,
+                                              double activ) const;
+
+        /** Calculate the muscle residual at static equilibrium.  The muscle
+        length is  static, the fiber velocity is zero, and the activation
+         derivative is zero.
+                @param projFibLen The projected fiber length
+                @param muscleLength The muscle length
+                @param activ The muscle activation
+        @returns Vector:
+                \li 0: muscle force residual
+                \li 1: derivative of residual wrt activation
+                \li 2: muscle force*/
+        SimTK::Vec3 calcFiberStaticEquilibResidual(double projFibLen,
+                                                   double muscleLength,
+                                                   double activ) const;
+
+
+
+
+        SimTK::Vector calcSolveMuscle(
+                SimTK::State& s, double activ, SimTK::Vector yDotInitialGuess)
+                                const;
+
+        class ImplicitSystemDerivativeSolver : public SimTK::OptimizerSystem {};
```

Comment:
In your case, this class can be completely contained in the cpp file; no need to expose it to the user. It doesn't need to be a nested class either.